python/cudf/cudf/_lib/csv.pyx

```diff
@@ -427,7 +427,11 @@ cpdef write_csv(
     cdef sink_info sink_info_c = make_sink_info(path_or_buf, data_sink_c)
 
     if header is True:
-        all_names = table._column_names
+        all_names = tuple(
+            table.columns.to_native_types(
+                na_rep=na_rep,
+            )
+        )
```

Comment:
Should we implement this without `.columns`? That constructs a Pandas Index object which we've seen to be kinda expensive.

Reply:
Changed the implementation to use `_column_names` instead of creating Index.